test/utils.ts

```diff
@@ -97,15 +99,41 @@ export function simplify(latex: string): string {
   return exprToString(engine.parse(latex)?.simplify());
 }
 
-export function checkJson(inExpr: SemiBoxedExpression | null): string {
+type ExprVariant =
+  | 'box'
+  | 'canonical'
+  | 'simplify'
+  | 'eval-auto'
+  | 'eval-mach'
+  | 'N-mach'
+  | 'N-auto';
+
+export function checkJson(
+  inExpr: SemiBoxedExpression | null,
+  variants?: Array<ExprVariant>
+): string {
   if (!inExpr) return 'null';
   try {
-    const precision = engine.precision;
-    engine.precision = 'auto';
-
-    const boxed = exprToString(engine.box(inExpr, { canonical: false }));
-
-    const expr = engine.box(inExpr);
+    const ce =
```

Comment:
As with one or two other locations within this file: this makes sense if wishing to use these utilities for expr's not parsed with the default (this file) test engine?

Reply:
Looks good.